src/components/shared/ProtectedRoutes/index.js

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import { Redirect, Route } from 'react-router';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { LOGIN_URL } from '../../../config/routing';
+import { getToken } from '../../../store/login/selectors';
+import { getType } from '../../../store/user/selectors';
+
+const BaseProtectedRouteComponent = ({
+  component: Component, token, userType, allowedUserType, ...rest
+}) => (
+  <Route
+    {...rest}
+    render={(props) => (
+      token && allowedUserType === userType
+        ? <Component {...props} />
+        : <Redirect to={LOGIN_URL} />)}
+  />
+);
+
+const mapStateToProps = (state) => ({
+  token: getToken(state),
+  userType: getType(state)
```

Comment:
Mb use some names like getUserType instead of getType. I think it could be a bit confusing in the future
It touches all selectors

Reply:
fixed